### PR TITLE
Extra-checks for some fields

### PR DESCRIPTION
**Pull request by miiu96**
- Added extra checks for some fields from transaction and smart contract result structure

### CODE REVIEW

File: process/elasticproc/converters/field.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// TruncateFieldIfExceedsMaxLength will truncate the provided field if the max length exceeds
+func TruncateFieldIfExceedsMaxLength(field string) string {
+	if len(field) > data.MaxFieldLength {
+		return field[:data.MaxFieldLength]
+	}
+
+	return field
+}
+
+// TruncateSliceElementsIfExceedsMaxLength will truncate the provided slice of field if the max length exceeds
```
Comment (bogdan-rosianu) [inline process/elasticproc/converters/field.go:14]: // TruncateSliceElementsIfExceedsMaxLength will truncate the provided slice of the field if the max length is exceeded

File: data/consts.go [hunk 2]
```diff
@@ -6,4 +6,6 @@ const (
 
 	// MaxESDTValueLength defines the maximum length for an ESDT value that can be parsed
 	MaxESDTValueLength = 100
+	// MaxFieldLength defines the maximum length for a keyword field
+	MaxFieldLength = 30000
```
Comment (iulianpascalau) [inline data/consts.go:10]: is this enough? How did we end up with this value?

File: process/elasticproc/accounts/accountsProcessor_test.go [hunk 3]
```diff
@@ -177,9 +177,10 @@ func TestGetESDTInfoNFTWithMetaData(t *testing.T) {
 	require.Equal(t, big.NewInt(1), balance)
 	require.Equal(t, "6f6b", prop)
 	require.Equal(t, &data.TokenMetaData{
-		Name:      nftName,
-		Creator:   creator,
-		Royalties: 2,
+		Name:       nftName,
+		Creator:    creator,
+		Royalties:  2,
+		Attributes: []byte{},
```
Comment (iulianpascalau) [inline process/elasticproc/accounts/accountsProcessor_test.go:183]: `make([]byte, 0)` instead of literal declaration

File: process/elasticproc/accounts/accountsProcessor_test.go [hunk 4]
```diff
@@ -343,7 +344,8 @@ func TestAccountsProcessor_PrepareAccountsMapESDT(t *testing.T) {
 		Properties:      "3032",
 		TokenNonce:      15,
 		Data: &data.TokenMetaData{
-			Creator: "creator",
+			Creator:    "creator",
+			Attributes: []byte{},
```
Comment (iulianpascalau) [inline process/elasticproc/accounts/accountsProcessor_test.go:348]: `make([]byte, 0)` instead of literal declaration

File: process/elasticproc/accounts/accountsProcessor_test.go [hunk 5]
```diff
@@ -357,7 +359,8 @@ func TestAccountsProcessor_PrepareAccountsMapESDT(t *testing.T) {
 		Properties:      "3032",
 		TokenNonce:      16,
 		Data: &data.TokenMetaData{
-			Creator: "creator",
+			Creator:    "creator",
+			Attributes: []byte{},
```
Comment (iulianpascalau) [inline process/elasticproc/accounts/accountsProcessor_test.go:363]: `make([]byte, 0)` instead of literal declaration

File: process/elasticproc/logsevents/nftsProcessor_test.go [hunk 6]
```diff
@@ -50,7 +50,8 @@ func TestNftsProcessor_processLogAndEventsNFTs(t *testing.T) {
 		Issuer:     "",
 		Nonce:      uint64(19),
 		Data: &data.TokenMetaData{
-			Creator: hex.EncodeToString([]byte("creator")),
+			Creator:    hex.EncodeToString([]byte("creator")),
+			Attributes: []byte{},
```
Comment (iulianpascalau) [inline process/elasticproc/logsevents/nftsProcessor_test.go:54]: `make([]byte, 0)` instead of literal declaration
This test (and also the other tests) should have worked with nil field. You can test them

File: process/elasticproc/accounts/accountsProcessor_test.go (hunk 3 above)
Comment (miiu96) [reply process/elasticproc/accounts/accountsProcessor_test.go:183]: done

File: process/elasticproc/accounts/accountsProcessor_test.go (hunk 4 above)
Comment (miiu96) [reply process/elasticproc/accounts/accountsProcessor_test.go:348]: done

File: process/elasticproc/accounts/accountsProcessor_test.go (hunk 5 above)
Comment (miiu96) [reply process/elasticproc/accounts/accountsProcessor_test.go:363]: done

File: process/elasticproc/logsevents/nftsProcessor_test.go (hunk 6 above)
Comment (miiu96) [reply process/elasticproc/logsevents/nftsProcessor_test.go:54]: done

File: process/elasticproc/converters/field.go (hunk 1 above)
Comment (miiu96) [reply process/elasticproc/converters/field.go]: done

**Review by bogdan-rosianu — APPROVED** [review]

File: process/elasticproc/transactions/transactionDBBuilder.go [hunk 7]
```diff
@@ -78,6 +78,8 @@ func (dtb *dbTransactionBuilder) prepareTransaction(
 		esdtValues = res.ESDTValues
 	}
 
+	senderUserName := converters.TruncateFieldIfExceedsMaxLength(string(tx.SndUserName))
```
Comment (iulianpascalau) [inline process/elasticproc/transactions/transactionDBBuilder.go:81]: not quite needed, we have an interceptor level protection set to 32 characters. Can remain like this for extra protection.

**Review by iulianpascalau — APPROVED** [review]
